### DISCUSSION TITLE
Scoring metrices for multiclass classification

### DISCUSSION THREAD

**Discussion by ashwinshetgaonkar** [rasbt/machine-learning-book#50] [Q&A]
on page 201,
It is mentioned that to use micro average instead of the macro (default one) fol. code is written:
pre_scorer = make_scorer(score_func=precision_score,
                        pos_label=1,
                        greater_is_better=True,
                        average='micro')
how to use this in multiclass ? (pos_label=1 is fine for binary problems)

**Comment by rasbt** [comment]
It would be like shown in the equation on page 201. Say you have 3 classes, it would compute 3 TP (true positive) scores:

1. TP_1 by treating class 1 as the TP and class 2 & 3 as "rest"
2. TP_2 by treating class 2 as the TP and class 1 & 3 as "rest"
3. TP_3 by treating class 3 as the TP and class 2 & 1 as "rest"

Then, you do the same thing for the FP (false positives). Essentially, it works the same way as macro-averaging, but instead of having multiple precision scores that you average, you have multiple TP and the FP and compute the precision score from the combined individual scores.
